src/types/routerPush.ts

```diff
@@ -1,21 +1,24 @@
 import { Routes } from '@/types/route'
 import { RoutesKey } from '@/types/routesMap'
 import { RouteParamsByKey } from '@/types/routeWithParams'
+import { RouteStateByKey } from '@/types/state'
 import { Url } from '@/types/url'
 import { AllPropertiesAreOptional } from '@/types/utilities'
 
-export type RouterPushOptions = {
+export type RouterPushOptions<
+  TState = unknown
+> = {
   query?: Record<string, string>,
   replace?: boolean,
+  state?: Partial<TState>,
```

Comment:
This is interesting. I didn't imaging you'd be able to set the state when navigating. But kinda cool that you can.